File: API_R_01Agosto2024/Isaac Alencar/q1_nomes.py
## QUESTÃo INCOMPLETA

def main():
    # perguntar o nome e ano de nascimento do usuário
    name = input('Qual o seu nome?:_')
    birth_year = int(input('Qual o ano do seu nascimento?:_ '))
    # mostrar idade
    age = calc_age(birth_year)
    print(age)
    # mostrar todos os divisores da idade dele
    print('Estes são os divisores da sua idade')
    print(get_age_divisors(age))
    # verificar o maior divisor primo e se houver, mostrar n vezes  n(numero primo)
    # caso nao houver, mostrar uma msg e retornar n vezes o número (n = media dos div da idade)

def calc_age(birth_year):
    current_year = 2024
    return current_year - birth_year

def get_age_divisors(age):
    i = 1
    count = 1
    result = 'Divisores: '
    while (count < age):
        if age % i == 0:
            divisor = str(age // i)
            result += divisor + ' '
        i += 1
        count += 1

    return result
      

def get_greater_prime_number(number):
    greater = 0

    if number >= greater:
        greater = number

    return greater



main()

File: API_R_01Agosto2024/Alanda Amabily/q1_nomes.js
import {question} from 'readline-sync'

function main(){
    const nome = question('Seu nome: ')
    const ano_nascimento = Number(question('Seu ano de nascimento: '))

    const idade = 2024 - ano_nascimento

    const divisores = mostrar_divisores(idade)
    const primo = calcular_num_primo(idade)

    console.log(`----------------
        divisores do numero ${idade}: ${divisores}
        é primo: ${primo}`)


}
function mostrar_divisores(n){
    console.log(`DIVISORES DE ${n}: `)
    let a = n;
    let divisores = '';

    while(a > 0){
        if(eh_divisor(n,a)){
            divisores += `${a}`
        }
    }
    console.log(divisores);

}
function eh_divisor(n, a){
    return n % a === 0;
}
function calcular_num_primo(numero){
    let i = 2;
    while(i != numero){
        if( numero % i === 0){
            i++;
            return 'sim';
        }else if(numero % i != 0){
            return 'nao';

        }
}
}
main()